swap/src/protocol/bob/event_loop.rs

```diff
@@ -117,11 +117,22 @@ impl EventLoop {
                                 let _ = responder.respond(*response);
                             }
                         }
-                        SwarmEvent::Behaviour(OutEvent::TransferProofReceived { msg, channel }) => {
-                            if msg.swap_id != self.swap_id {
+                        SwarmEvent::Behaviour(OutEvent::TransferProofReceived { msg, channel, peer }) => {
+                            let swap_id = msg.swap_id;
+
+                            if peer != self.alice_peer_id {
+                                tracing::warn!(
+                                            %swap_id,
+                                            "Ignoring malicious transfer proof from {}, expected to receive it from {}",
+                                            peer,
+                                            self.alice_peer_id);
+                                        continue;
+                            }
+
+                            if swap_id != self.swap_id {
```

Comment:
We keep loading up the event loop with these kind of fairly low-level checks when really, we should only be delegating to other components because this functionality in here is quite hard to test (as we can see by the lack of unit tests for this functionality).

Probably about time to create dedicated `NetworkBehaviour`s that encapsulate this kind of logic and only emit events to the `EventLoop` if they actually need processing.

Reply:
I totally agree, recorded this ticket https://github.com/comit-network/xmr-btc-swap/issues/458 for now. Planning to do some refactoring after we are mainnet ready :)